dsdl_compiler/libcanard_dsdl_compiler/code_type_template.tmpl

```diff
@@ -250,12 +262,25 @@ int32_t ${type_name}_decode_internal(
     // Static array (${f.name})
     for (c = 0; c < ${f.array_size}; c++)
     {
+                    %if isinstance(f.data_type.element_type, CompoundType)
+        offset = ${f.c_type}_decode_internal(transfer,
```

Comment:
Same here, seems like a temporary is needed which is then added to offset unless it's negative (error).